### ISSUE TITLE
Intelligent and more correct Html `addCssClass`

### ISSUE THREAD

**Issue by kartik-v** [yiisoft/yii2#7896]
I will be submitting a PR for improving and making the `Html::addCssClass` more intelligent and more practical to use for all use cases - similar or better than jquery addClass.

## Enhancements:

- Parse the original `$options['class']` correctly irrespective of whitespaces  (tabs, newlines or spaces)
- Parse the `$class` param correctly irrespective of whitespaces  (tabs, newlines or spaces)
- If the `$class` property is separated by spaces... it will validate each class separately already added and 
- Intelligently do not append css class if it already exists irrespective of the whitespace

### Example
```php
$options = ['class' => '    kartik']; // note the whitespace
Html::addCssClass($options, ' hide    kartik  robot');
Html::addCssClass($options, "\tkartik   hide fancy    box     holy");
```
### Output
```php
// generated output
$options['class'] =  '    kartik  hide    kartik  robot         kartik   hide fancy    box     holy';
```

### Example
```php
$options = ['class' => '    kartik']; // note the whitespace
Html::addCssClass($options, ' hide    kartik  robot');
Html::addCssClass($options, "\tkartik   hide fancy    box     holy");
```
### Output
```php
// generated output
$options['class'] =  'kartik hide robot fancy box holy';
```

**Comment by cebe** [comment]
how realistic is it that this method will receive unclean input?

**Comment by kartik-v** [comment]
The purpose is not just for unclean input... but it currently does not support adding CSS class separated by white spaces. Let's say I want to add 5 different CSS classes to an `options`... the current approach is to call this method 5 times.

**Comment by cebe** [comment]
sure, that should be possible.

**Comment by klimov-paul** [comment]
> how realistic is it that this method will receive unclean input? In general improvement is good but it should not add too much overhead.

Agreed.

> Let's say I want to add 5 different CSS classes to an options... the current approach is to call this method 5 times.

Will be resolved by #8458

Closed in favor of #6043
